src/nlsolve/nlsolve.jl

```diff
@@ -31,7 +31,12 @@ function nlsolve!(nlsolver::AbstractNLSolver, integrator, cache=nothing, repeat_
 
   local ndz
   for iter in 1:maxiters
-    always_new && update_W!(nlsolver, integrator, cache, γW, repeat_step, (true, true))
+    if always_new
+        @. nlsolver.cache.ustep = integrator.uprev + nlsolver.γ * nlsolver.z
```

Comment:
This is a bit wasteful because the same `nlsolver.cache.ustep` will be evaluated later. I'll handle this later.